src/command/mod.rs

```diff
@@ -1,32 +1,90 @@
+use std::time::Duration;
+
+use tokio::sync::oneshot;
+
 use crate::codec::Token;
+use crate::server::Context;
+use crate::storage::StorageCommand;
 
 mod command;
 pub use command::{Command, CommandError};
 
 /// CommandProcessor is responsible for taking a group of tokens, executing them,
 /// and returning the result.
-pub struct CommandProcessor {}
+pub struct CommandProcessor {
+    context: Context,
+}
 
 impl CommandProcessor {
-    pub fn execute_command(&self, command: &Command) -> Vec<Token> {
+    pub fn new(context: Context) -> Self {
+        Self { context }
+    }
+
+    pub async fn execute_command(&self, command: &Command) -> Vec<Token> {
```

Comment:
I don't see a code path that returns more (or less) than one token. Do you expect to in the future?

Reply:
Yep! This will return multiple tokens when results are arrays, such as the `KEYS` command.